.github/workflows/pre-commit.yaml

```
@@ -0,0 +1,28 @@
+name: pre-commit
+
+on:
+  push:
+    branches: [main]
+  pull_request:
```

To keep it consistent with the other workflows let's run this when pushing to any branch:
```suggestion
on: [push, pull_request]
```